packages/appinio_swiper/lib/enums.dart

```diff
@@ -1,53 +1,129 @@
-enum AppinioSwiperState {
-  swipe,
-  swipeLeft,
-  swipeRight,
-  unswipe,
-  swipeUp,
-  swipeDown
+import 'package:flutter/material.dart';
+
+sealed class SwiperActivity extends Tween<Offset> {
+  SwiperActivity(
+    this.animation, {
+    required super.begin,
+    required super.end,
+  });
+
+  final AnimationController animation;
+
+  Offset get currentOffset => evaluate(animation);
+
+  AxisDirection get direction {
+    return (end! - begin!).toAxisDirection();
+  }
+}
+
+class Swipe extends SwiperActivity {
+  Swipe(
+    super.animation, {
+    required super.begin,
+    required super.end,
+  });
+}
+
+class Unswipe extends SwiperActivity {
+  Unswipe(
+    super.animation, {
+    required super.begin,
+  }) : super(end: Offset.zero);
+
+  @override
+  AxisDirection get direction => super.direction.opposite;
+}
+
+class CancelSwipe extends SwiperActivity {
+  CancelSwipe(
+    super.animation, {
+    required super.begin,
+  }) : super(end: Offset.zero);
+
+  @override
+  AxisDirection get direction => super.direction.opposite;
 }
 
-enum AppinioSwiperDirection { none, left, right, top, bottom }
+class DrivenActivity extends SwiperActivity {
+  DrivenActivity(
+    super.animation, {
+    required this.curve,
+    required super.begin,
+    required super.end,
+  });
 
-class AppinioSwipeOptions {
-  final bool top;
-  final bool bottom;
+  final Curve curve;
+
+  late final Animation<double> curvedAnimation = CurvedAnimation(
+    parent: animation,
+    curve: curve,
+  );
+
+  @override
+  Offset get currentOffset => evaluate(curvedAnimation);
+}
+
+class SwipeOptions {
+  final bool up;
+  final bool down;
   final bool left;
   final bool right;
-  final bool horizontal;
-  final bool vertical;
-  final bool allDirections;
-
-  const AppinioSwipeOptions.symmetric({
-    this.horizontal = false,
-    this.vertical = false,
-  })  : left = false,
-        top = false,
-        right = false,
-        bottom = false,
-        allDirections =
-            !((horizontal && !vertical) || (!horizontal && vertical));
 
-  /// If horizontal and vertical both are true/false then simply make the allDirections true.
+  const SwipeOptions.symmetric({
```

Comment:
Replaced redundant variables with getters/mirrored `EdgeInsets`.

Also, I removed the `Appinio` from the front here and in a couple other places because:
1. it doesn't contribute any information so it's unnecessarily verbose
2. Appinio is hard to type, and if everything is prefixed with it you have to type it out before you can auto complete. It was driving me nuts XD

If you really want me to I can roll this back but I will be very sad ):